traversal/walk.go

```diff
@@ -155,6 +155,32 @@ func (prog Progress) walkAdv(n datamodel.Node, s selector.Selector, fn AdvVisitF
 		}
 		prog.Budget.NodeBudget--
 	}
+
+	// refiy the node if advised.
+	if rs, ok := s.(selector.Reifiable); ok {
+		adl := rs.NamedReifier()
+		if prog.Cfg.LinkSystem.KnownReifiers == nil {
```

Comment:
do we need to treat a nil KnownReifiers as a special case? it really isn't any different from a non-nil map with zero entries, if we're just reading.

if you really want to capture the "no known reifiers" case, I'd suggest `len(...) == 0`. the easiest approach is to just let it get to the "not supported" error below.

Reply:
i think this check is here because deference of nil map can panic